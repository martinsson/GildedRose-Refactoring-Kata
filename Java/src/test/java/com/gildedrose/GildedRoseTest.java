package com.gildedrose;

import static org.junit.Assert.*;

import org.approvaltests.legacycode.LegacyApprovals;
import org.approvaltests.reporters.DiffReporter;
import org.approvaltests.reporters.UseReporter;
import org.approvaltests.reporters.macosx.DiffMergeReporter;
import org.approvaltests.reporters.macosx.KDiff3Reporter;
import org.junit.Test;

public class GildedRoseTest {

    @Test
    public void lockDown() throws Exception {
        String[] itemNames = {"foo", "Aged Brie", "Backstage passes to a TAFKAL80ETC concert", "Sulfuras, Hand of Ragnaros"};
        Object[] qualities = {0, 1, 3, 47, 48, 49, 50, 51};
        Object[] sellIns = {0, 4, 5, 6, 11, 12};
        Object[][] variations = {itemNames, sellIns, qualities};
        LegacyApprovals.LockDown(this, "testOneItem", variations);
    }

    public String testOneItem(String itemName, Integer sellIn, Integer quality) {
        Item[] items = new Item[] { new Item(itemName, sellIn, quality) };
        GildedRose app = new GildedRose(items);
        app.updateQuality();
        Item item = app.items[0];
        return item.toString();
    }

}
