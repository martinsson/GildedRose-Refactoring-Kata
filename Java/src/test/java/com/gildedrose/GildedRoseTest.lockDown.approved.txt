[foo, 0, 0] = foo, -1, 0 
[Aged Brie, 0, 0] = Aged Brie, -1, 2 
[Backstage passes to a TAFKAL80ETC concert, 0, 0] = Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Sulfuras, Hand of Ragnaros, 0, 0] = Sulfuras, Hand of Ragnaros, 0, 0 
[foo, 4, 0] = foo, 3, 0 
[Aged Brie, 4, 0] = Aged Brie, 3, 1 
[Backstage passes to a TAFKAL80ETC concert, 4, 0] = Backstage passes to a TAFKAL80ETC concert, 3, 3 
[Sulfuras, Hand of Ragnaros, 4, 0] = Sulfuras, Hand of Ragnaros, 4, 0 
[foo, 5, 0] = foo, 4, 0 
[Aged Brie, 5, 0] = Aged Brie, 4, 1 
[Backstage passes to a TAFKAL80ETC concert, 5, 0] = Backstage passes to a TAFKAL80ETC concert, 4, 3 
[Sulfuras, Hand of Ragnaros, 5, 0] = Sulfuras, Hand of Ragnaros, 5, 0 
[foo, 6, 0] = foo, 5, 0 
[Aged Brie, 6, 0] = Aged Brie, 5, 1 
[Backstage passes to a TAFKAL80ETC concert, 6, 0] = Backstage passes to a TAFKAL80ETC concert, 5, 2 
[Sulfuras, Hand of Ragnaros, 6, 0] = Sulfuras, Hand of Ragnaros, 6, 0 
[foo, 11, 0] = foo, 10, 0 
[Aged Brie, 11, 0] = Aged Brie, 10, 1 
[Backstage passes to a TAFKAL80ETC concert, 11, 0] = Backstage passes to a TAFKAL80ETC concert, 10, 1 
[Sulfuras, Hand of Ragnaros, 11, 0] = Sulfuras, Hand of Ragnaros, 11, 0 
[foo, 12, 0] = foo, 11, 0 
[Aged Brie, 12, 0] = Aged Brie, 11, 1 
[Backstage passes to a TAFKAL80ETC concert, 12, 0] = Backstage passes to a TAFKAL80ETC concert, 11, 1 
[Sulfuras, Hand of Ragnaros, 12, 0] = Sulfuras, Hand of Ragnaros, 12, 0 
[foo, 0, 1] = foo, -1, 0 
[Aged Brie, 0, 1] = Aged Brie, -1, 3 
[Backstage passes to a TAFKAL80ETC concert, 0, 1] = Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Sulfuras, Hand of Ragnaros, 0, 1] = Sulfuras, Hand of Ragnaros, 0, 1 
[foo, 4, 1] = foo, 3, 0 
[Aged Brie, 4, 1] = Aged Brie, 3, 2 
[Backstage passes to a TAFKAL80ETC concert, 4, 1] = Backstage passes to a TAFKAL80ETC concert, 3, 4 
[Sulfuras, Hand of Ragnaros, 4, 1] = Sulfuras, Hand of Ragnaros, 4, 1 
[foo, 5, 1] = foo, 4, 0 
[Aged Brie, 5, 1] = Aged Brie, 4, 2 
[Backstage passes to a TAFKAL80ETC concert, 5, 1] = Backstage passes to a TAFKAL80ETC concert, 4, 4 
[Sulfuras, Hand of Ragnaros, 5, 1] = Sulfuras, Hand of Ragnaros, 5, 1 
[foo, 6, 1] = foo, 5, 0 
[Aged Brie, 6, 1] = Aged Brie, 5, 2 
[Backstage passes to a TAFKAL80ETC concert, 6, 1] = Backstage passes to a TAFKAL80ETC concert, 5, 3 
[Sulfuras, Hand of Ragnaros, 6, 1] = Sulfuras, Hand of Ragnaros, 6, 1 
[foo, 11, 1] = foo, 10, 0 
[Aged Brie, 11, 1] = Aged Brie, 10, 2 
[Backstage passes to a TAFKAL80ETC concert, 11, 1] = Backstage passes to a TAFKAL80ETC concert, 10, 2 
[Sulfuras, Hand of Ragnaros, 11, 1] = Sulfuras, Hand of Ragnaros, 11, 1 
[foo, 12, 1] = foo, 11, 0 
[Aged Brie, 12, 1] = Aged Brie, 11, 2 
[Backstage passes to a TAFKAL80ETC concert, 12, 1] = Backstage passes to a TAFKAL80ETC concert, 11, 2 
[Sulfuras, Hand of Ragnaros, 12, 1] = Sulfuras, Hand of Ragnaros, 12, 1 
[foo, 0, 3] = foo, -1, 1 
[Aged Brie, 0, 3] = Aged Brie, -1, 5 
[Backstage passes to a TAFKAL80ETC concert, 0, 3] = Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Sulfuras, Hand of Ragnaros, 0, 3] = Sulfuras, Hand of Ragnaros, 0, 3 
[foo, 4, 3] = foo, 3, 2 
[Aged Brie, 4, 3] = Aged Brie, 3, 4 
[Backstage passes to a TAFKAL80ETC concert, 4, 3] = Backstage passes to a TAFKAL80ETC concert, 3, 6 
[Sulfuras, Hand of Ragnaros, 4, 3] = Sulfuras, Hand of Ragnaros, 4, 3 
[foo, 5, 3] = foo, 4, 2 
[Aged Brie, 5, 3] = Aged Brie, 4, 4 
[Backstage passes to a TAFKAL80ETC concert, 5, 3] = Backstage passes to a TAFKAL80ETC concert, 4, 6 
[Sulfuras, Hand of Ragnaros, 5, 3] = Sulfuras, Hand of Ragnaros, 5, 3 
[foo, 6, 3] = foo, 5, 2 
[Aged Brie, 6, 3] = Aged Brie, 5, 4 
[Backstage passes to a TAFKAL80ETC concert, 6, 3] = Backstage passes to a TAFKAL80ETC concert, 5, 5 
[Sulfuras, Hand of Ragnaros, 6, 3] = Sulfuras, Hand of Ragnaros, 6, 3 
[foo, 11, 3] = foo, 10, 2 
[Aged Brie, 11, 3] = Aged Brie, 10, 4 
[Backstage passes to a TAFKAL80ETC concert, 11, 3] = Backstage passes to a TAFKAL80ETC concert, 10, 4 
[Sulfuras, Hand of Ragnaros, 11, 3] = Sulfuras, Hand of Ragnaros, 11, 3 
[foo, 12, 3] = foo, 11, 2 
[Aged Brie, 12, 3] = Aged Brie, 11, 4 
[Backstage passes to a TAFKAL80ETC concert, 12, 3] = Backstage passes to a TAFKAL80ETC concert, 11, 4 
[Sulfuras, Hand of Ragnaros, 12, 3] = Sulfuras, Hand of Ragnaros, 12, 3 
[foo, 0, 47] = foo, -1, 45 
[Aged Brie, 0, 47] = Aged Brie, -1, 49 
[Backstage passes to a TAFKAL80ETC concert, 0, 47] = Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Sulfuras, Hand of Ragnaros, 0, 47] = Sulfuras, Hand of Ragnaros, 0, 47 
[foo, 4, 47] = foo, 3, 46 
[Aged Brie, 4, 47] = Aged Brie, 3, 48 
[Backstage passes to a TAFKAL80ETC concert, 4, 47] = Backstage passes to a TAFKAL80ETC concert, 3, 50 
[Sulfuras, Hand of Ragnaros, 4, 47] = Sulfuras, Hand of Ragnaros, 4, 47 
[foo, 5, 47] = foo, 4, 46 
[Aged Brie, 5, 47] = Aged Brie, 4, 48 
[Backstage passes to a TAFKAL80ETC concert, 5, 47] = Backstage passes to a TAFKAL80ETC concert, 4, 50 
[Sulfuras, Hand of Ragnaros, 5, 47] = Sulfuras, Hand of Ragnaros, 5, 47 
[foo, 6, 47] = foo, 5, 46 
[Aged Brie, 6, 47] = Aged Brie, 5, 48 
[Backstage passes to a TAFKAL80ETC concert, 6, 47] = Backstage passes to a TAFKAL80ETC concert, 5, 49 
[Sulfuras, Hand of Ragnaros, 6, 47] = Sulfuras, Hand of Ragnaros, 6, 47 
[foo, 11, 47] = foo, 10, 46 
[Aged Brie, 11, 47] = Aged Brie, 10, 48 
[Backstage passes to a TAFKAL80ETC concert, 11, 47] = Backstage passes to a TAFKAL80ETC concert, 10, 48 
[Sulfuras, Hand of Ragnaros, 11, 47] = Sulfuras, Hand of Ragnaros, 11, 47 
[foo, 12, 47] = foo, 11, 46 
[Aged Brie, 12, 47] = Aged Brie, 11, 48 
[Backstage passes to a TAFKAL80ETC concert, 12, 47] = Backstage passes to a TAFKAL80ETC concert, 11, 48 
[Sulfuras, Hand of Ragnaros, 12, 47] = Sulfuras, Hand of Ragnaros, 12, 47 
[foo, 0, 48] = foo, -1, 46 
[Aged Brie, 0, 48] = Aged Brie, -1, 50 
[Backstage passes to a TAFKAL80ETC concert, 0, 48] = Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Sulfuras, Hand of Ragnaros, 0, 48] = Sulfuras, Hand of Ragnaros, 0, 48 
[foo, 4, 48] = foo, 3, 47 
[Aged Brie, 4, 48] = Aged Brie, 3, 49 
[Backstage passes to a TAFKAL80ETC concert, 4, 48] = Backstage passes to a TAFKAL80ETC concert, 3, 50 
[Sulfuras, Hand of Ragnaros, 4, 48] = Sulfuras, Hand of Ragnaros, 4, 48 
[foo, 5, 48] = foo, 4, 47 
[Aged Brie, 5, 48] = Aged Brie, 4, 49 
[Backstage passes to a TAFKAL80ETC concert, 5, 48] = Backstage passes to a TAFKAL80ETC concert, 4, 50 
[Sulfuras, Hand of Ragnaros, 5, 48] = Sulfuras, Hand of Ragnaros, 5, 48 
[foo, 6, 48] = foo, 5, 47 
[Aged Brie, 6, 48] = Aged Brie, 5, 49 
[Backstage passes to a TAFKAL80ETC concert, 6, 48] = Backstage passes to a TAFKAL80ETC concert, 5, 50 
[Sulfuras, Hand of Ragnaros, 6, 48] = Sulfuras, Hand of Ragnaros, 6, 48 
[foo, 11, 48] = foo, 10, 47 
[Aged Brie, 11, 48] = Aged Brie, 10, 49 
[Backstage passes to a TAFKAL80ETC concert, 11, 48] = Backstage passes to a TAFKAL80ETC concert, 10, 49 
[Sulfuras, Hand of Ragnaros, 11, 48] = Sulfuras, Hand of Ragnaros, 11, 48 
[foo, 12, 48] = foo, 11, 47 
[Aged Brie, 12, 48] = Aged Brie, 11, 49 
[Backstage passes to a TAFKAL80ETC concert, 12, 48] = Backstage passes to a TAFKAL80ETC concert, 11, 49 
[Sulfuras, Hand of Ragnaros, 12, 48] = Sulfuras, Hand of Ragnaros, 12, 48 
[foo, 0, 49] = foo, -1, 47 
[Aged Brie, 0, 49] = Aged Brie, -1, 50 
[Backstage passes to a TAFKAL80ETC concert, 0, 49] = Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Sulfuras, Hand of Ragnaros, 0, 49] = Sulfuras, Hand of Ragnaros, 0, 49 
[foo, 4, 49] = foo, 3, 48 
[Aged Brie, 4, 49] = Aged Brie, 3, 50 
[Backstage passes to a TAFKAL80ETC concert, 4, 49] = Backstage passes to a TAFKAL80ETC concert, 3, 50 
[Sulfuras, Hand of Ragnaros, 4, 49] = Sulfuras, Hand of Ragnaros, 4, 49 
[foo, 5, 49] = foo, 4, 48 
[Aged Brie, 5, 49] = Aged Brie, 4, 50 
[Backstage passes to a TAFKAL80ETC concert, 5, 49] = Backstage passes to a TAFKAL80ETC concert, 4, 50 
[Sulfuras, Hand of Ragnaros, 5, 49] = Sulfuras, Hand of Ragnaros, 5, 49 
[foo, 6, 49] = foo, 5, 48 
[Aged Brie, 6, 49] = Aged Brie, 5, 50 
[Backstage passes to a TAFKAL80ETC concert, 6, 49] = Backstage passes to a TAFKAL80ETC concert, 5, 50 
[Sulfuras, Hand of Ragnaros, 6, 49] = Sulfuras, Hand of Ragnaros, 6, 49 
[foo, 11, 49] = foo, 10, 48 
[Aged Brie, 11, 49] = Aged Brie, 10, 50 
[Backstage passes to a TAFKAL80ETC concert, 11, 49] = Backstage passes to a TAFKAL80ETC concert, 10, 50 
[Sulfuras, Hand of Ragnaros, 11, 49] = Sulfuras, Hand of Ragnaros, 11, 49 
[foo, 12, 49] = foo, 11, 48 
[Aged Brie, 12, 49] = Aged Brie, 11, 50 
[Backstage passes to a TAFKAL80ETC concert, 12, 49] = Backstage passes to a TAFKAL80ETC concert, 11, 50 
[Sulfuras, Hand of Ragnaros, 12, 49] = Sulfuras, Hand of Ragnaros, 12, 49 
[foo, 0, 50] = foo, -1, 48 
[Aged Brie, 0, 50] = Aged Brie, -1, 50 
[Backstage passes to a TAFKAL80ETC concert, 0, 50] = Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Sulfuras, Hand of Ragnaros, 0, 50] = Sulfuras, Hand of Ragnaros, 0, 50 
[foo, 4, 50] = foo, 3, 49 
[Aged Brie, 4, 50] = Aged Brie, 3, 50 
[Backstage passes to a TAFKAL80ETC concert, 4, 50] = Backstage passes to a TAFKAL80ETC concert, 3, 50 
[Sulfuras, Hand of Ragnaros, 4, 50] = Sulfuras, Hand of Ragnaros, 4, 50 
[foo, 5, 50] = foo, 4, 49 
[Aged Brie, 5, 50] = Aged Brie, 4, 50 
[Backstage passes to a TAFKAL80ETC concert, 5, 50] = Backstage passes to a TAFKAL80ETC concert, 4, 50 
[Sulfuras, Hand of Ragnaros, 5, 50] = Sulfuras, Hand of Ragnaros, 5, 50 
[foo, 6, 50] = foo, 5, 49 
[Aged Brie, 6, 50] = Aged Brie, 5, 50 
[Backstage passes to a TAFKAL80ETC concert, 6, 50] = Backstage passes to a TAFKAL80ETC concert, 5, 50 
[Sulfuras, Hand of Ragnaros, 6, 50] = Sulfuras, Hand of Ragnaros, 6, 50 
[foo, 11, 50] = foo, 10, 49 
[Aged Brie, 11, 50] = Aged Brie, 10, 50 
[Backstage passes to a TAFKAL80ETC concert, 11, 50] = Backstage passes to a TAFKAL80ETC concert, 10, 50 
[Sulfuras, Hand of Ragnaros, 11, 50] = Sulfuras, Hand of Ragnaros, 11, 50 
[foo, 12, 50] = foo, 11, 49 
[Aged Brie, 12, 50] = Aged Brie, 11, 50 
[Backstage passes to a TAFKAL80ETC concert, 12, 50] = Backstage passes to a TAFKAL80ETC concert, 11, 50 
[Sulfuras, Hand of Ragnaros, 12, 50] = Sulfuras, Hand of Ragnaros, 12, 50 
[foo, 0, 51] = foo, -1, 49 
[Aged Brie, 0, 51] = Aged Brie, -1, 51 
[Backstage passes to a TAFKAL80ETC concert, 0, 51] = Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Sulfuras, Hand of Ragnaros, 0, 51] = Sulfuras, Hand of Ragnaros, 0, 51 
[foo, 4, 51] = foo, 3, 50 
[Aged Brie, 4, 51] = Aged Brie, 3, 51 
[Backstage passes to a TAFKAL80ETC concert, 4, 51] = Backstage passes to a TAFKAL80ETC concert, 3, 51 
[Sulfuras, Hand of Ragnaros, 4, 51] = Sulfuras, Hand of Ragnaros, 4, 51 
[foo, 5, 51] = foo, 4, 50 
[Aged Brie, 5, 51] = Aged Brie, 4, 51 
[Backstage passes to a TAFKAL80ETC concert, 5, 51] = Backstage passes to a TAFKAL80ETC concert, 4, 51 
[Sulfuras, Hand of Ragnaros, 5, 51] = Sulfuras, Hand of Ragnaros, 5, 51 
[foo, 6, 51] = foo, 5, 50 
[Aged Brie, 6, 51] = Aged Brie, 5, 51 
[Backstage passes to a TAFKAL80ETC concert, 6, 51] = Backstage passes to a TAFKAL80ETC concert, 5, 51 
[Sulfuras, Hand of Ragnaros, 6, 51] = Sulfuras, Hand of Ragnaros, 6, 51 
[foo, 11, 51] = foo, 10, 50 
[Aged Brie, 11, 51] = Aged Brie, 10, 51 
[Backstage passes to a TAFKAL80ETC concert, 11, 51] = Backstage passes to a TAFKAL80ETC concert, 10, 51 
[Sulfuras, Hand of Ragnaros, 11, 51] = Sulfuras, Hand of Ragnaros, 11, 51 
[foo, 12, 51] = foo, 11, 50 
[Aged Brie, 12, 51] = Aged Brie, 11, 51 
[Backstage passes to a TAFKAL80ETC concert, 12, 51] = Backstage passes to a TAFKAL80ETC concert, 11, 51 
[Sulfuras, Hand of Ragnaros, 12, 51] = Sulfuras, Hand of Ragnaros, 12, 51 
